pkg/observer/controller.go

```diff
@@ -243,6 +247,63 @@ func (c *controller) dropInProgressNodeGroups(nodeGroups v1.NodeGroupList, cnrs
 	return restingNodeGroups
 }
 
+// get the cluster-autoscaler last scaleUp activity time
+func stringToTime(s string) (time.Time, error) {
+	sec, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return time.Unix(sec, 0), nil
+}
+
+// query cluster-autoscaler metrics to figure out if it's safe to start a new CNR
+func (c *controller) safeToStartCycle() bool {
+	safeToStart := true
```

Comment:
suggest removing this variable and just using `return true` / `return false` directly for each branch. Easier to follow and make changes without tracking what the variable state is.

Reply:
👍